appeals/api/src/server/endpoints/appellant-cases/appellant-cases.service.js

```diff
@@ -94,6 +95,46 @@ const updateAppellantCaseValidationOutcome = async (
 		}
 	}
 
+	const updatedAppeal = await appealRepository.getAppealById(Number(appealId));
+	``;
+	if (isOutcomeIncomplete(validationOutcome.name) && updatedAppeal) {
+		const recipientEmail = appeal.agent?.email || appeal.appellant?.email;
+		if (!recipientEmail) {
+			throw new Error(ERROR_NO_RECIPIENT_EMAIL);
+		}
+
+		const newIncompleteReasons =
+			updatedAppeal.appellantCase?.appellantCaseIncompleteReasonsOnAppellantCases;
+		if (!newIncompleteReasons) {
+			throw new Error(ERROR_NOT_FOUND);
+		}
+
+		const incompleteReasonsList = newIncompleteReasons.flatMap((item) => {
```

Comment:
Could we consider abstracting the logic building incompleteReasonsList into a reusable function?

Reply:
Yes this definitely be fun to convert into email nested bullet points

Reply:
and this is a brilliant solution, of flattening them and wrap the text reasons with the parent...

Reply:
I'm thinking to refactor it when I do the invalid ticket, as that's where the reuse becomes relevant. What you think?

Reply:
> and this is a brilliant solution, of flattening them and wrap the text reasons with the parent...

I will unashamedly admit I had chatGPT assist me on that one ;)